test/dub_package/source/app.d

```diff
@@ -43,3 +43,119 @@ unittest
     scope parser = new Parser!ASTBase(null, null, false);
     assert(parser !is null);
 }
+
+
+// inits the DMD compiler
+void init()
+{
+    import ddmd.globals : global;
+
+    global._init;
+    global.params.isLinux = 1;
+    import ddmd.dmodule : Module;
+    import ddmd.id : Id;
+    import ddmd.mtype : Type;
+    import ddmd.target : Target;
+    import ddmd.expression : Expression;
+    import ddmd.objc : Objc;
+    import ddmd.builtin : builtin_init;
+
+    // Initialization
+    Type._init();
+    Id.initialize();
+    Module._init();
+    Target._init();
+    Expression._init();
+    Objc._init();
+    builtin_init();
+}
```

Comment:
Is there any chance we can move a generic initialization function into the DMD codebase?

Reply:
Currently this is handled in `mars.d`, which contains the `main` function of DMD. This file should not be included in the Dub package, but I don't see why we can't move this to a separate function, called `initFrontEnd` which can be called both from the compiler and the Dub package. Perhaps we can add a new file with a shared constructor that is only part of the Dub package that calls this `initFrontEnd` function.

Reply:
> Perhaps we can add a new file with a shared constructor that is only part of the Dub package that calls this initFrontEnd function.

Yeah, I think this would be the best way to move forward. This file could also contain other convenience, modern D wrapper. How about `ddmd/libdmd.d`, `ddmd/library.d`, or `ddmd/frontend.d`?

Reply:
I don't like `ddmd/libdmd.d`, the others are fine, I think I prefer `ddmd/frontend.d`. But I don't want to limit ourselves to this file. I think we should look at what we want to add and add it to the modules where it makes most sense. I don't mind adding several new modules and or packages if that makes more sense.